mesonbuild/optinterpreter.py

```diff
@@ -199,10 +201,9 @@ def func_option(self, args: T.Tuple[str], kwargs: 'FuncOptionArgs') -> None:
 
         # Drop the arguments we've already consumed
         n_kwargs = {k: v for k, v in kwargs.items()
-                    if k not in {'type', 'description', 'deprecated'}}
+                    if k not in {'type', 'description', 'deprecated', 'yield'}}
 
-        opt = parser(description, n_kwargs, kwargs['yield'])
-        opt.deprecated = kwargs['deprecated']
+        opt = parser(description, (kwargs['yield'], kwargs['deprecated']), n_kwargs)
```

Comment:
Neat trick, but... on re-reading  this I think we can't have an intermediate step here? The logic from the earlier commit is now n_kwargs at position 2 of 3, and typed_kwargs (really get_callee_args) goes nuts over this, because what it *actually* wants is for kwargs to be *last*.

Reply:
In other words, this errored out for me during bisection.